llama_index/llms/openai.py

```diff
@@ -181,12 +186,9 @@ def _get_all_kwargs(self, **kwargs: Any) -> Dict[str, Any]:
         return {**self._credential_kwargs, **self._model_kwargs, **kwargs}
 
     def _chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
-        if not self._is_chat_model:
```

Comment:
These validations are unnecessary, as callers of this private method already checked for it in their own conditionals